janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java

```diff
@@ -1288,15 +1302,35 @@ public GraphDatabaseConfiguration(ReadConfiguration localConfig) {
                 } else {
                     log.info("Using configured timestamp provider {}", localBasicConfiguration.get(TIMESTAMP_PROVIDER));
                 }
-
                 globalWrite.freezeConfiguration();
             } else {
+                final boolean allowUpgrade;
+
+                if (localBasicConfiguration.has(ALLOW_UPGRADE))
+                    allowUpgrade = localBasicConfiguration.get(ALLOW_UPGRADE);
+                else if (globalWrite.has(ALLOW_UPGRADE))
+                    allowUpgrade = globalWrite.get(ALLOW_UPGRADE);
+                else
+                    allowUpgrade = ALLOW_UPGRADE.getDefaultValue();
+
+                globalWrite.set(ALLOW_UPGRADE, allowUpgrade);
+
+                if (!allowUpgrade) {
+		    try {
+                        if (globalWrite.has(INITIAL_STORAGE_VERSION) && globalWrite.get(INITIAL_STORAGE_VERSION).equals(JanusGraphConstants.STORAGE_VERSION)) {
```

Comment:
I don't think we're particularly interested in the common case where `INITIAL_STORAGE_VERSION == STORAGE_VERSION`, so no need to log it. Just handle other case.